src/Project/BackgroundGridAPI.jl

```diff
@@ -0,0 +1,263 @@
+#=
+ MIT License
+
+ Copyright (c) 2010-present David A. Kopriva and other contributors: AUTHORS.md
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
+ --- End License
+=#
+
+"""
+    addBackgroundGrid(proj::Project, bgSize::Array{Float64})
+
+Add the background grid block with the grid size to be a 3-vector. Use this when there
+is an outer boundary defined in the model.
+"""
+function addBackgroundGrid!(proj::Project, bgSize::Array{Float64})
+    disableUndo()
+    disableNotifications()
+    setBackgroundGridSize!(proj, bgSize, "background grid size")
+    enableUndo()
+    registerWithUndoManager(proj,removeBackgroundGrid!,(nothing,),"Add Background Grid")
+    enableNotifications()
+    postNotificationWithName(proj,"BGRID_DID_CHANGE_NOTIFICATION",(nothing,))
+end
+"""
+    addBackgroundGrid!(proj::Project, box::Array{Float64},  N::Array{Int} )
+
+Add the background grid block with bounding box = [TOP, LEFT, BOTTOM, RIGHT]
+and the number of intervals in each diredction. Use this when there
+is _no_ outer boundary defined in the model.
+"""
+function addBackgroundGrid!(proj::Project, box::Array{Float64}, N::Array{Int})
+    disableUndo()
+    disableNotifications()
+    proj.bounds     = box
+    proj.userBounds = deepcopy(box)
+
+    dx          = zeros(Float64,3)
```

Comment:
If you are always working with arrays of size `3`, you could consider using StaticArrays.jl for efficiency.

Reply:
StaticArrays.jl is not part of Julia, and hence one is again relying on the kindness of strangers. Performance is not an issue here, and should not be traded for code stability.

Reply:
I stand by my position that one writes only standard code, using standard libraries if necessary (e.g. STL). it is tempting to use something else, but code has to outlive the author.